src/cahillKeyes.js

```diff
@@ -0,0 +1,713 @@
+      xy = interpolate(l, lT, jointE(m), jointT(m));
+    } else {
+      // point is in middle segment
+      l = l - lT;
+      xy = interpolate(l, lM, jointT(m), jointF(m));
+    }
+    return xy;
+  }
+
+  function zoneL(m, p, l15) {
+    var p73a = parallel73(m),
+      p73 = p73a.parallel73,
+      lT = lengthTorridSegment(m),
+      lM = lengthMiddleSegment(m),
+      xy,
+      lF = p73a.lengthParallel73,
+      l = l15 + (p - 15) * (lT + lM + lF - l15) / 58;
+    if (l <= lT) {
+      //on torrid segment
+      xy = interpolate(l, lT, jointE(m), jointF(m));
+    } else if (l <= lT + lM) {
+      //on middle segment
+      l = l - lT;
+      xy = interpolate(l, lM, jointT(m), jointF(m));
+    } else {
+      //on frigid segment
+      l = l - lT - lM;
+      xy = interpolate(l, lF, jointF(m), p73);
+    }
+    return xy;
+  }
+
+  // convert half-octant meridian,parallel to x,y coordinates.
+  // arguments are meridian, parallel
+
+  function mp2xy(m, p) {
+    var xy = [0, 0],
+      lT,
+      p15a,
+      p15,
+      flag15,
+      l15;
+
+    if (m === 0) {
+      // zones (a) and (b)
+      if (p >= 75) {
+        xy = zoneA(m, p);
+      } else {
+        xy = zoneB(m, p);
+      }
+    } else if (p >= 75) {
+      xy = zoneC(m, p);
+    } else if (p === 0) {
+      xy = zoneD(m, p);
+    } else if (p >= 73 && m <= 30) {
+      xy = zoneE(m, p);
+    } else if (m === 45) {
+      if (p <= 15) {
+        xy = zoneF(m, p);
+      } else if (p <= 73) {
+        xy = zoneG(m, p);
+      } else {
+        xy = zoneH(m, p);
+      }
+    } else {
+      if (m <= 29) {
+        xy = zoneI(m, p);
+      } else {
+        // supple zones (j), (k) and (l)
+        if (p >= 73) {
+          xy = zoneJ(m, p);
+        } else {
+          //zones (k) and (l)
+          p15a = circleLineIntersection(
+            CK.pointC,
+            CK.radius,
+            jointT(m),
+            jointF(m)
+          );
+          flag15 = p15a[0];
+          p15 = p15a[1];
+          lT = lengthTorridSegment(m);
+          if (flag15 === 1) {
+            // intersection is in middle segment
+            l15 = lT + distance(jointT(m), p15);
+          } else {
+            // intersection is in torrid segment
+            p15a = circleLineIntersection(
+              CK.pointC,
+              CK.radius,
+              jointE(m),
+              jointT(m)
+            );
+            flag15 = p15a[0];
+            p15 = p15a[1];
+            if (flag15 === 0) {
+              //console.log("Something weird!");
+              // TODO: Trap this! Something odd happened!
+            }
+            l15 = lT - distance(jointT(m), p15);
+          }
+          if (p <= 15) {
+            xy = zoneK(m, p, l15);
+          } else {
+            //zone (l)
+            xy = zoneL(m, p, l15);
+          }
+        }
+      }
+    }
+    return xy;
+  }
+
+  // from half-octant to megamap (single rotated octant)
+
+  function mj2g(xy, octant) {
```

Comment:
This is taken directly from MaryJo implementation but to have it in the spirit of polyhedral map projection, all work of translating/rotating would really need to go in `faceProjection` L17...but I personally have never really had the time and stamina to go thru it...

Reply:
Yes and I suppose this is what we'll need to do if we want to manipulate the tree of faces for the Antarctica challenge.